themes/src/main/resources/theme/base/login/login-password.ftl

```diff
@@ -10,10 +10,16 @@
                     <div class="${properties.kcFormGroupClass!} no-bottom-margin">
                         <hr/>
                         <label for="password" class="${properties.kcLabelClass!}">${msg("password")}</label>
-                        <input tabindex="2" id="password" class="${properties.kcInputClass!}" name="password"
-                               type="password" autocomplete="on" autofocus
-                               aria-invalid="<#if messagesPerField.existsError('password')>true</#if>"
-                        />
+                        <div class="${properties.kcInputGroup!}">
+                            <input tabindex="2" id="password" class="${properties.kcInputClass!}" name="password"
+                                   type="password" autocomplete="on" autofocus
+                                   aria-invalid="<#if messagesPerField.existsError('password')>true</#if>"
+                            />
+                            <button class="pf-c-button pf-m-control" type="button" aria-label="Show password"
```

Comment:
All the human readable text, such as the `aria-label` attribute, should be localized.

Reply:
I agree. How would you change the `aria-label` on the `onclick` event with localized texts? I won't be able to access the `msg()` function from the js module. Of course I could pass the messages as parameter like this: 
```
<button onclick="passwordVisibility.toggle(this, 'password', '${msg('showPassword')}', '${msg('hidePassword')}')">
```

Another (probably worse) idea is a separat (global) script block in the FTL and access them in the module directly: 
```
<script>
    const msg = {
        hidePassword: "${msg('hidePassword')}",
        showPassword: "${msg('showPassword')}",
    };
</script>
<button onclick="passwordVisibility.toggle(this, 'password')">
```

Reply:
I think the best approach would possibly be to use `data-` attributes to set this. It's a bit of an old school method, but it should work. So the HTML would look something like:

```html
<button data-label-show="${msg('showPassword')}" data-label-hide="${msg('showPassword')}">
```

You can learn more about working with data attributes on [MDN](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes).

Reply:
I like it :) You can check the new commit.